sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -81,6 +81,9 @@ trait ProgressReporter extends Logging {
   /** Holds the most recent query progress updates.  Accesses must lock on the queue itself. */
   private val progressBuffer = new mutable.Queue[StreamingQueryProgress]()
 
+  /** Count the total input records of this streaming query. */
```

Comment:
I would rather use this space (I meant one of fields for summary of streaming query) to represent other thing. The number of total input records would be less and less important once the query has been running for days or even weeks. The latency of batch would be more considerable one, as it has been considered as critical aspect of streaming, even applied to the micro-batch.

Reply:
Add a new class `QuerySummary`. We can add additional summary information about the query. Currently I only add a 'totalInputRecords' information.

Reply:
I'm revisiting around "query summary" - so that's only the status for entire batches, as it doesn't belong to stream progress. I'm still on the fence of adding this unless we have clear value of having this - we are having two different paths on retrieving information due to just retrieve `total input records`.